vcs/git/repo.go

```diff
@@ -235,11 +233,19 @@ func (r *Repository) Commits(opt vcs.CommitsOptions) ([]*vcs.Commit, uint, error
 			commits = append(commits, r.makeCommit(c))
 		}
 		total++
-		return total < uint(MaxCommits)
```

Comment:
I think it's still a good idea to keep this for now, so we don't regress the perf of any clients that do not pass the `NoTotal parameter`.